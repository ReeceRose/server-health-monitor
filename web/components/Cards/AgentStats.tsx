type Props = {
  title: string;
  subtitle: string;
  iconColour: string;
};

const AgentStatsCard: React.FC<Props> = ({ title, subtitle, iconColour }) => {
  return (
    <div className="relative flex flex-col min-w-0 mb-6 break-words bg-white rounded shadow-lg xl:mb-0">
      <div className="flex-auto p-4">
        <div className="flex flex-wrap">
          <div className="relative flex-1 flex-grow w-full max-w-full pr-4">
            <h5 className="text-xs font-bold text-gray-400 uppercase">
              {subtitle}
            </h5>
            <span className="text-xl font-semibold text-gray-700">{title}</span>
          </div>
          <div className="relative flex-initial w-auto pl-4">
            <div
              className={
                'text-white p-3 text-center inline-flex items-center justify-center w-12 h-12 shadow-lg rounded-full ' +
                iconColour
              }
            >
              <i className="fas fa-server"></i>
            </div>
          </div>
        </div>
      </div>
    </div>
  );
};

export default AgentStatsCard;
