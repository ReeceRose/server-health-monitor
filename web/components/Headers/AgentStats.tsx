import AgentStats from '../Cards/AgentStats';

export default function AgentStatus(): JSX.Element {
  return (
    <div className="relative pt-12 pb-32 bg-gray-800 md:pt-32">
      <div className="w-full px-4 mx-auto md:px-10">
        <div>
          <div className="flex flex-wrap">
            <div className="w-full px-4 lg:w-6/12 xl:w-3/12">
              <AgentStats
                subtitle="ACTIVE SERVERS"
                title="2"
                iconColour="bg-green-400"
              />
            </div>
            <div className="w-full px-4 lg:w-6/12 xl:w-3/12">
              <AgentStats
                subtitle="INACTIVE SERVERS"
                title="2"
                iconColour="bg-red-400"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  );
}
