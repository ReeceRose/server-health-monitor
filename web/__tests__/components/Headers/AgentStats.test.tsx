import { render } from '@testing-library/react';

import AgentStats from '../../../components/Headers/AgentStats';

test('AgentStats component matches snapshot', () => {
  const { asFragment } = render(<AgentStats />);
  expect(asFragment()).toMatchSnapshot();
});
