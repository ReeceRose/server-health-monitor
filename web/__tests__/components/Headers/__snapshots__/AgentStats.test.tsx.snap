// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AgentStats component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="relative pt-12 pb-32 bg-gray-800 md:pt-32"
  >
    <div
      class="w-full px-4 mx-auto md:px-10"
    >
      <div>
        <div
          class="flex flex-wrap"
        >
          <div
            class="w-full px-4 lg:w-6/12 xl:w-3/12"
          >
            <div
              class="relative flex flex-col min-w-0 mb-6 break-words bg-white rounded shadow-lg xl:mb-0"
            >
              <div
                class="flex-auto p-4"
              >
                <div
                  class="flex flex-wrap"
                >
                  <div
                    class="relative flex-1 flex-grow w-full max-w-full pr-4"
                  >
                    <h5
                      class="text-xs font-bold text-gray-400 uppercase"
                    >
                      ACTIVE SERVERS
                    </h5>
                    <span
                      class="text-xl font-semibold text-gray-700"
                    >
                      2
                    </span>
                  </div>
                  <div
                    class="relative flex-initial w-auto pl-4"
                  >
                    <div
                      class="text-white p-3 text-center inline-flex items-center justify-center w-12 h-12 shadow-lg rounded-full bg-green-400"
                    >
                      <i
                        class="fas fa-server"
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="w-full px-4 lg:w-6/12 xl:w-3/12"
          >
            <div
              class="relative flex flex-col min-w-0 mb-6 break-words bg-white rounded shadow-lg xl:mb-0"
            >
              <div
                class="flex-auto p-4"
              >
                <div
                  class="flex flex-wrap"
                >
                  <div
                    class="relative flex-1 flex-grow w-full max-w-full pr-4"
                  >
                    <h5
                      class="text-xs font-bold text-gray-400 uppercase"
                    >
                      INACTIVE SERVERS
                    </h5>
                    <span
                      class="text-xl font-semibold text-gray-700"
                    >
                      2
                    </span>
                  </div>
                  <div
                    class="relative flex-initial w-auto pl-4"
                  >
                    <div
                      class="text-white p-3 text-center inline-flex items-center justify-center w-12 h-12 shadow-lg rounded-full bg-red-400"
                    >
                      <i
                        class="fas fa-server"
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
